Sources/Datadog/Core/Persistence/Reading/Reader.swift

```diff
@@ -7,7 +7,7 @@
 import Foundation
 
 internal struct Batch {
-    let data: Data
+    let payloads: [Data]
```

Comment:
Similar comment on using "payloads" name. Here the problem is even more visible, as we're mixing domains - `Batch` doesn't need to know that it participates data uploads, so calling it `events: [Data]` could be even more desired here. I also wish we have a doc comment on this property as it's pretty crucial. WDYT?